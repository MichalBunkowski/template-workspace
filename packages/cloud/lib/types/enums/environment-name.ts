export enum EnvironmentName {
  Develop = 'deve',
  Production = 'prod',
}
