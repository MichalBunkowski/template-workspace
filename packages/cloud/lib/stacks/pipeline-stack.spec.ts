import * as cdk from '@aws-cdk/core';

import { EnvironmentName } from '../types/enums/environment-name';
import { PipelineStack } from './pipeline.stack';

describe('PipelineStack', () => {
  it('should match snapshot', () => {
    const app = new cdk.App();
    // WHEN
    const stack = new PipelineStack(app, 'MyTestStack');
    // THEN
    const actual = app.synth().getStackArtifact(stack.artifactId).template;
    expect(actual.Resources ?? {}).toMatchSnapshot();
  });
});
