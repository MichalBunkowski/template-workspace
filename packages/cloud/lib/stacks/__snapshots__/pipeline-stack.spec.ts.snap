// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelineStack should match snapshot 1`] = `
Object {
  "Pipeline9850B417": Object {
    "DependsOn": Array [
      "PipelineRoleDefaultPolicy7BDC1ABB",
      "PipelineRoleB27FAA37",
    ],
    "Properties": Object {
      "ArtifactStore": Object {
        "Location": Object {
          "Ref": "PipelineArtifactsBucketAEA9A052",
        },
        "Type": "S3",
      },
      "Name": "CdkAppPipeline",
      "RestartExecutionOnUpdate": true,
      "RoleArn": Object {
        "Fn::GetAtt": Array [
          "PipelineRoleB27FAA37",
          "Arn",
        ],
      },
      "Stages": Array [
        Object {
          "Actions": Array [
            Object {
              "ActionTypeId": Object {
                "Category": "Source",
                "Owner": "ThirdParty",
                "Provider": "GitHub",
                "Version": "1",
              },
              "Configuration": Object {
                "Branch": "main",
                "OAuthToken": "{{resolve:secretsmanager:github_token:SecretString:::}}",
                "Owner": "MichalBunkowski",
                "PollForSourceChanges": false,
                "Repo": "template-workspace",
              },
              "Name": "MichalBunkowski_template-workspace",
              "OutputArtifacts": Array [
                Object {
                  "Name": "MichalBunkowski_template_workspace_Source",
                },
              ],
              "RunOrder": 1,
            },
          ],
          "Name": "Source",
        },
        Object {
          "Actions": Array [
            Object {
              "ActionTypeId": Object {
                "Category": "Build",
                "Owner": "AWS",
                "Provider": "CodeBuild",
                "Version": "1",
              },
              "Configuration": Object {
                "EnvironmentVariables": "[{\\"name\\":\\"_PROJECT_CONFIG_HASH\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"64fea05f86122c343a5af78e6fd5be8dc5fc5eec6a14f18bfe83d914f0a26813\\"}]",
                "ProjectName": Object {
                  "Ref": "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                },
              },
              "InputArtifacts": Array [
                Object {
                  "Name": "MichalBunkowski_template_workspace_Source",
                },
              ],
              "Name": "Synth",
              "OutputArtifacts": Array [
                Object {
                  "Name": "Synth_Output",
                },
              ],
              "RoleArn": Object {
                "Fn::GetAtt": Array [
                  "PipelineBuildSynthCodePipelineActionRole4E7A6C97",
                  "Arn",
                ],
              },
              "RunOrder": 1,
            },
          ],
          "Name": "Build",
        },
        Object {
          "Actions": Array [
            Object {
              "ActionTypeId": Object {
                "Category": "Build",
                "Owner": "AWS",
                "Provider": "CodeBuild",
                "Version": "1",
              },
              "Configuration": Object {
                "EnvironmentVariables": "[{\\"name\\":\\"_PROJECT_CONFIG_HASH\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"38f1eeb22c590a6fe1573cfa845971b035360bb1b4c1954a18d5d88ad36b255b\\"}]",
                "ProjectName": Object {
                  "Ref": "PipelineUpdatePipelineSelfMutationDAA41400",
                },
              },
              "InputArtifacts": Array [
                Object {
                  "Name": "Synth_Output",
                },
              ],
              "Name": "SelfMutate",
              "RoleArn": Object {
                "Fn::GetAtt": Array [
                  "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleD6D4E5CF",
                  "Arn",
                ],
              },
              "RunOrder": 1,
            },
          ],
          "Name": "UpdatePipeline",
        },
        Object {
          "Actions": Array [
            Object {
              "ActionTypeId": Object {
                "Category": "Deploy",
                "Owner": "AWS",
                "Provider": "CloudFormation",
                "Version": "1",
              },
              "Configuration": Object {
                "ActionMode": "CHANGE_SET_REPLACE",
                "Capabilities": "CAPABILITY_NAMED_IAM,CAPABILITY_AUTO_EXPAND",
                "ChangeSetName": "PipelineChange",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PipelinePreProdPrepareRole1603ABD3",
                    "Arn",
                  ],
                },
                "StackName": "PreProd-AppStack",
                "TemplatePath": "Synth_Output::assembly-MyTestStack-PreProd/PreProd-AppStack.template.json",
              },
              "InputArtifacts": Array [
                Object {
                  "Name": "Synth_Output",
                },
              ],
              "Name": "Prepare",
              "RoleArn": Object {
                "Fn::GetAtt": Array [
                  "PipelinePreProdPrepareCodePipelineActionRole514E9351",
                  "Arn",
                ],
              },
              "RunOrder": 1,
            },
            Object {
              "ActionTypeId": Object {
                "Category": "Deploy",
                "Owner": "AWS",
                "Provider": "CloudFormation",
                "Version": "1",
              },
              "Configuration": Object {
                "ActionMode": "CHANGE_SET_EXECUTE",
                "ChangeSetName": "PipelineChange",
                "StackName": "PreProd-AppStack",
              },
              "Name": "Deploy",
              "RoleArn": Object {
                "Fn::GetAtt": Array [
                  "PipelinePreProdDeployCodePipelineActionRole75F74DD5",
                  "Arn",
                ],
              },
              "RunOrder": 2,
            },
          ],
          "Name": "PreProd",
        },
      ],
    },
    "Type": "AWS::CodePipeline::Pipeline",
  },
  "PipelineArtifactsBucketAEA9A052": Object {
    "DeletionPolicy": "Retain",
    "Properties": Object {
      "BucketEncryption": Object {
        "ServerSideEncryptionConfiguration": Array [
          Object {
            "ServerSideEncryptionByDefault": Object {
              "SSEAlgorithm": "aws:kms",
            },
          },
        ],
      },
      "PublicAccessBlockConfiguration": Object {
        "BlockPublicAcls": true,
        "BlockPublicPolicy": true,
        "IgnorePublicAcls": true,
        "RestrictPublicBuckets": true,
      },
    },
    "Type": "AWS::S3::Bucket",
    "UpdateReplacePolicy": "Retain",
  },
  "PipelineBuildSynthCdkBuildProject6BEFA8E6": Object {
    "Properties": Object {
      "Artifacts": Object {
        "Type": "CODEPIPELINE",
      },
      "EncryptionKey": "alias/aws/s3",
      "Environment": Object {
        "ComputeType": "BUILD_GENERAL1_SMALL",
        "Image": "aws/codebuild/standard:5.0",
        "ImagePullCredentialsType": "CODEBUILD",
        "PrivilegedMode": false,
        "Type": "LINUX_CONTAINER",
      },
      "ServiceRole": Object {
        "Fn::GetAtt": Array [
          "PipelineBuildSynthCdkBuildProjectRole231EEA2A",
          "Arn",
        ],
      },
      "Source": Object {
        "BuildSpec": "{
  \\"version\\": \\"0.2\\",
  \\"phases\\": {
    \\"build\\": {
      \\"commands\\": [
        \\"yarn bootstrap\\",
        \\"yarn synth\\"
      ]
    }
  },
  \\"artifacts\\": {
    \\"base-directory\\": \\"packages/cloud/cdk.out\\",
    \\"files\\": \\"**/*\\"
  }
}",
        "Type": "CODEPIPELINE",
      },
    },
    "Type": "AWS::CodeBuild::Project",
  },
  "PipelineBuildSynthCdkBuildProjectRole231EEA2A": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "Service": "codebuild.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
  "PipelineBuildSynthCdkBuildProjectRoleDefaultPolicyFB6C941C": Object {
    "Properties": Object {
      "PolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": Array [
              "logs:CreateLogGroup",
              "logs:CreateLogStream",
              "logs:PutLogEvents",
            ],
            "Effect": "Allow",
            "Resource": Array [
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:/aws/codebuild/",
                    Object {
                      "Ref": "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                    },
                  ],
                ],
              },
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:/aws/codebuild/",
                    Object {
                      "Ref": "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                    },
                    ":*",
                  ],
                ],
              },
            ],
          },
          Object {
            "Action": Array [
              "codebuild:CreateReportGroup",
              "codebuild:CreateReport",
              "codebuild:UpdateReport",
              "codebuild:BatchPutTestCases",
              "codebuild:BatchPutCodeCoverages",
            ],
            "Effect": "Allow",
            "Resource": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "arn:",
                  Object {
                    "Ref": "AWS::Partition",
                  },
                  ":codebuild:",
                  Object {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  Object {
                    "Ref": "AWS::AccountId",
                  },
                  ":report-group/",
                  Object {
                    "Ref": "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                  },
                  "-*",
                ],
              ],
            },
          },
          Object {
            "Action": Array [
              "s3:GetObject*",
              "s3:GetBucket*",
              "s3:List*",
              "s3:DeleteObject*",
              "s3:PutObject*",
              "s3:Abort*",
            ],
            "Effect": "Allow",
            "Resource": Array [
              Object {
                "Fn::GetAtt": Array [
                  "PipelineArtifactsBucketAEA9A052",
                  "Arn",
                ],
              },
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "PipelineArtifactsBucketAEA9A052",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            ],
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "PipelineBuildSynthCdkBuildProjectRoleDefaultPolicyFB6C941C",
      "Roles": Array [
        Object {
          "Ref": "PipelineBuildSynthCdkBuildProjectRole231EEA2A",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
  "PipelineBuildSynthCodePipelineActionRole4E7A6C97": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "AWS": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":iam::",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":root",
                  ],
                ],
              },
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
  "PipelineBuildSynthCodePipelineActionRoleDefaultPolicy92C90290": Object {
    "Properties": Object {
      "PolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": Array [
              "codebuild:BatchGetBuilds",
              "codebuild:StartBuild",
              "codebuild:StopBuild",
            ],
            "Effect": "Allow",
            "Resource": Object {
              "Fn::GetAtt": Array [
                "PipelineBuildSynthCdkBuildProject6BEFA8E6",
                "Arn",
              ],
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "PipelineBuildSynthCodePipelineActionRoleDefaultPolicy92C90290",
      "Roles": Array [
        Object {
          "Ref": "PipelineBuildSynthCodePipelineActionRole4E7A6C97",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
  "PipelinePreProdDeployCodePipelineActionRole75F74DD5": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "AWS": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":iam::",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":root",
                  ],
                ],
              },
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
  "PipelinePreProdDeployCodePipelineActionRoleDefaultPolicyFF397D9A": Object {
    "Properties": Object {
      "PolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": Array [
              "cloudformation:DescribeChangeSet",
              "cloudformation:DescribeStacks",
              "cloudformation:ExecuteChangeSet",
            ],
            "Condition": Object {
              "StringEqualsIfExists": Object {
                "cloudformation:ChangeSetName": "PipelineChange",
              },
            },
            "Effect": "Allow",
            "Resource": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "arn:",
                  Object {
                    "Ref": "AWS::Partition",
                  },
                  ":cloudformation:",
                  Object {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  Object {
                    "Ref": "AWS::AccountId",
                  },
                  ":stack/PreProd-AppStack/*",
                ],
              ],
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "PipelinePreProdDeployCodePipelineActionRoleDefaultPolicyFF397D9A",
      "Roles": Array [
        Object {
          "Ref": "PipelinePreProdDeployCodePipelineActionRole75F74DD5",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
  "PipelinePreProdPrepareCodePipelineActionRole514E9351": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "AWS": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":iam::",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":root",
                  ],
                ],
              },
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
  "PipelinePreProdPrepareCodePipelineActionRoleDefaultPolicyE94FE1EB": Object {
    "Properties": Object {
      "PolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "iam:PassRole",
            "Effect": "Allow",
            "Resource": Object {
              "Fn::GetAtt": Array [
                "PipelinePreProdPrepareRole1603ABD3",
                "Arn",
              ],
            },
          },
          Object {
            "Action": Array [
              "s3:GetObject*",
              "s3:GetBucket*",
              "s3:List*",
            ],
            "Effect": "Allow",
            "Resource": Array [
              Object {
                "Fn::GetAtt": Array [
                  "PipelineArtifactsBucketAEA9A052",
                  "Arn",
                ],
              },
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "PipelineArtifactsBucketAEA9A052",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            ],
          },
          Object {
            "Action": Array [
              "cloudformation:CreateChangeSet",
              "cloudformation:DeleteChangeSet",
              "cloudformation:DescribeChangeSet",
              "cloudformation:DescribeStacks",
            ],
            "Condition": Object {
              "StringEqualsIfExists": Object {
                "cloudformation:ChangeSetName": "PipelineChange",
              },
            },
            "Effect": "Allow",
            "Resource": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "arn:",
                  Object {
                    "Ref": "AWS::Partition",
                  },
                  ":cloudformation:",
                  Object {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  Object {
                    "Ref": "AWS::AccountId",
                  },
                  ":stack/PreProd-AppStack/*",
                ],
              ],
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "PipelinePreProdPrepareCodePipelineActionRoleDefaultPolicyE94FE1EB",
      "Roles": Array [
        Object {
          "Ref": "PipelinePreProdPrepareCodePipelineActionRole514E9351",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
  "PipelinePreProdPrepareRole1603ABD3": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "Service": "cloudformation.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
  "PipelinePreProdPrepareRoleDefaultPolicyF136FB19": Object {
    "Properties": Object {
      "PolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": Array [
              "s3:GetObject*",
              "s3:GetBucket*",
              "s3:List*",
            ],
            "Effect": "Allow",
            "Resource": Array [
              Object {
                "Fn::GetAtt": Array [
                  "PipelineArtifactsBucketAEA9A052",
                  "Arn",
                ],
              },
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "PipelineArtifactsBucketAEA9A052",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            ],
          },
          Object {
            "Action": "*",
            "Effect": "Allow",
            "Resource": "*",
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "PipelinePreProdPrepareRoleDefaultPolicyF136FB19",
      "Roles": Array [
        Object {
          "Ref": "PipelinePreProdPrepareRole1603ABD3",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
  "PipelineRoleB27FAA37": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "Service": "codepipeline.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
  "PipelineRoleDefaultPolicy7BDC1ABB": Object {
    "Properties": Object {
      "PolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": Array [
              "s3:GetObject*",
              "s3:GetBucket*",
              "s3:List*",
              "s3:DeleteObject*",
              "s3:PutObject*",
              "s3:Abort*",
            ],
            "Effect": "Allow",
            "Resource": Array [
              Object {
                "Fn::GetAtt": Array [
                  "PipelineArtifactsBucketAEA9A052",
                  "Arn",
                ],
              },
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "PipelineArtifactsBucketAEA9A052",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            ],
          },
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Resource": Object {
              "Fn::GetAtt": Array [
                "PipelineBuildSynthCodePipelineActionRole4E7A6C97",
                "Arn",
              ],
            },
          },
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Resource": Object {
              "Fn::GetAtt": Array [
                "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleD6D4E5CF",
                "Arn",
              ],
            },
          },
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Resource": Object {
              "Fn::GetAtt": Array [
                "PipelinePreProdPrepareCodePipelineActionRole514E9351",
                "Arn",
              ],
            },
          },
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Resource": Object {
              "Fn::GetAtt": Array [
                "PipelinePreProdDeployCodePipelineActionRole75F74DD5",
                "Arn",
              ],
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "PipelineRoleDefaultPolicy7BDC1ABB",
      "Roles": Array [
        Object {
          "Ref": "PipelineRoleB27FAA37",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
  "PipelineSourceMichalBunkowskitemplateworkspaceWebhookResource035D227D": Object {
    "Properties": Object {
      "Authentication": "GITHUB_HMAC",
      "AuthenticationConfiguration": Object {
        "SecretToken": "{{resolve:secretsmanager:github_token:SecretString:::}}",
      },
      "Filters": Array [
        Object {
          "JsonPath": "$.ref",
          "MatchEquals": "refs/heads/{Branch}",
        },
      ],
      "RegisterWithThirdParty": true,
      "TargetAction": "MichalBunkowski_template-workspace",
      "TargetPipeline": Object {
        "Ref": "Pipeline9850B417",
      },
      "TargetPipelineVersion": 1,
    },
    "Type": "AWS::CodePipeline::Webhook",
  },
  "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleD6D4E5CF": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "AWS": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":iam::",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":root",
                  ],
                ],
              },
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
  "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleDefaultPolicyE626265B": Object {
    "Properties": Object {
      "PolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": Array [
              "codebuild:BatchGetBuilds",
              "codebuild:StartBuild",
              "codebuild:StopBuild",
            ],
            "Effect": "Allow",
            "Resource": Object {
              "Fn::GetAtt": Array [
                "PipelineUpdatePipelineSelfMutationDAA41400",
                "Arn",
              ],
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleDefaultPolicyE626265B",
      "Roles": Array [
        Object {
          "Ref": "PipelineUpdatePipelineSelfMutateCodePipelineActionRoleD6D4E5CF",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
  "PipelineUpdatePipelineSelfMutationDAA41400": Object {
    "Properties": Object {
      "Artifacts": Object {
        "Type": "CODEPIPELINE",
      },
      "EncryptionKey": "alias/aws/s3",
      "Environment": Object {
        "ComputeType": "BUILD_GENERAL1_SMALL",
        "Image": "aws/codebuild/standard:5.0",
        "ImagePullCredentialsType": "CODEBUILD",
        "PrivilegedMode": false,
        "Type": "LINUX_CONTAINER",
      },
      "Name": "CdkAppPipeline-selfupdate",
      "ServiceRole": Object {
        "Fn::GetAtt": Array [
          "PipelineUpdatePipelineSelfMutationRole57E559E8",
          "Arn",
        ],
      },
      "Source": Object {
        "BuildSpec": "{
  \\"version\\": \\"0.2\\",
  \\"phases\\": {
    \\"install\\": {
      \\"commands\\": [
        \\"npm install -g aws-cdk\\"
      ]
    },
    \\"build\\": {
      \\"commands\\": [
        \\"cdk -a . deploy MyTestStack --require-approval=never --verbose\\"
      ]
    }
  }
}",
        "Type": "CODEPIPELINE",
      },
    },
    "Type": "AWS::CodeBuild::Project",
  },
  "PipelineUpdatePipelineSelfMutationRole57E559E8": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "Service": "codebuild.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
  "PipelineUpdatePipelineSelfMutationRoleDefaultPolicyA225DA4E": Object {
    "Properties": Object {
      "PolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": Array [
              "logs:CreateLogGroup",
              "logs:CreateLogStream",
              "logs:PutLogEvents",
            ],
            "Effect": "Allow",
            "Resource": Array [
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:/aws/codebuild/",
                    Object {
                      "Ref": "PipelineUpdatePipelineSelfMutationDAA41400",
                    },
                  ],
                ],
              },
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:/aws/codebuild/",
                    Object {
                      "Ref": "PipelineUpdatePipelineSelfMutationDAA41400",
                    },
                    ":*",
                  ],
                ],
              },
            ],
          },
          Object {
            "Action": Array [
              "codebuild:CreateReportGroup",
              "codebuild:CreateReport",
              "codebuild:UpdateReport",
              "codebuild:BatchPutTestCases",
              "codebuild:BatchPutCodeCoverages",
            ],
            "Effect": "Allow",
            "Resource": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "arn:",
                  Object {
                    "Ref": "AWS::Partition",
                  },
                  ":codebuild:",
                  Object {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  Object {
                    "Ref": "AWS::AccountId",
                  },
                  ":report-group/",
                  Object {
                    "Ref": "PipelineUpdatePipelineSelfMutationDAA41400",
                  },
                  "-*",
                ],
              ],
            },
          },
          Object {
            "Action": "sts:AssumeRole",
            "Condition": Object {
              "ForAnyValue:StringEquals": Object {
                "iam:ResourceTag/aws-cdk:bootstrap-role": Array [
                  "image-publishing",
                  "file-publishing",
                  "deploy",
                ],
              },
            },
            "Effect": "Allow",
            "Resource": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "arn:*:iam::",
                  Object {
                    "Ref": "AWS::AccountId",
                  },
                  ":role/*",
                ],
              ],
            },
          },
          Object {
            "Action": "cloudformation:DescribeStacks",
            "Effect": "Allow",
            "Resource": "*",
          },
          Object {
            "Action": "s3:ListBucket",
            "Effect": "Allow",
            "Resource": "*",
          },
          Object {
            "Action": Array [
              "s3:GetObject*",
              "s3:GetBucket*",
              "s3:List*",
            ],
            "Effect": "Allow",
            "Resource": Array [
              Object {
                "Fn::GetAtt": Array [
                  "PipelineArtifactsBucketAEA9A052",
                  "Arn",
                ],
              },
              Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "PipelineArtifactsBucketAEA9A052",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            ],
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "PipelineUpdatePipelineSelfMutationRoleDefaultPolicyA225DA4E",
      "Roles": Array [
        Object {
          "Ref": "PipelineUpdatePipelineSelfMutationRole57E559E8",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
}
`;
